### ISSUE TITLE
Add a way to claim names with optional accounts

### ISSUE THREAD

**Issue by ChillerDragon** [ddnet-insta/ddnet-insta#187]
The requests of elo systems and similar keep coming up. There are already casual stats but it does not seem to be enough for everyone. So classical chat accounts (or maybe its wise to wait for ddnet account system rollout) that can be signed in into optionally. Accounts can be used to claim a name. If a player joins with a name that has an account attached the player will be renamed to ``(1)player`` if they login they get their name ``player`` unlocked.

This is just a thought. I am not sure if I like it yet. Accounts are in the way of the casual nature of how pvp games are played. They create a barrier. And even optional accounts be annoying. I can already see the entire server running around with a ``(1)`` prefix because either they are too lazy to login or forgot their password or forgot that accounts exist.

Also the question is how to integrate the current stats with the accounts one. Are the /top commands shared? It could display both type of names with account lock on them and without in one ranking list. And the ones with account lock are then protected from being griefed with bad stats such as dieing or losing. But they could technically still be outranked by someone without an account.

The question is should it even be tied to the in game name. Players change their names all the time. Does that require a new account? A migration?

A big plus of implementing a account system is that it can be used by all the ddnet-insta based modifications. Such as pvp level mods.

**Comment by PathosEthosLogos** [comment]
Yeah, I think there is an account system WIP for official DDNet. There was another thread, but here is one https://github.com/ddnet/ddnet/issues/3411#issuecomment-1534189301

Apparently, it's global for all mods. This will also be tied to Steam (but not exclusively, I think).



There was another solution that was implemented by Teerace, but requires some coding for the server. You press a bind, which does an RCon token login. Register for an account through the website. I'm assuming it's not easy to copy/paste their solution though.

**Comment by ChillerDragon** [comment]
I begin to like the idea of optional accounts. And simple ones without ddnet yet because the ddnet account system feels a bit too immature.

Accounts should be fully disconnected from stats. So players without accounts can farm stats and get ranks in the exact same way. The only thing accounts allow to do is use the ``/claim_name`` chat command. Which claims the name they are currently using. This way users can keep one account (username/password) and even change the display name they claim if they wish to do so. Claimed names can only be used by the logged in account owner this way the name based stats can be protected if users want to do so. If the player never used the ``/claim_name`` command this account locks no name. A account can still be useful if the config ``sv_require_login_to_vote`` is added to combat fun voting of newly joined players that can not even figure out or do not even bother to create an account.

Before rolling this out it might require a system to lock names for manual approval. For example nobody can claim names with high amounts of points without an admin confirming it. Or nobody can claim names from a admin provided list of high profile players.

The whole accounts thing it self needs a separate issue to track which features it should include and how the database columns should look like. Thats here now: https://github.com/ddnet-insta/ddnet-insta/issues/236

**Comment by ChillerDragon** [comment]
Renaming the player to (1)player after join is problematic because now the player can grief the name during the claim lookup time. So all players should be (1)player until their name is approved. Which can look a bit ugly. So the question is how and when is that feature deactivated? Should names still be locked if accounts are currently off? Should there be a config to disable the name claim feature that is unrelated to accounts being enabled?
